contracts/Test/TestRibbonStakingRewards.sol

```diff
@@ -124,18 +124,22 @@ contract KovanStakingRewards is
 
   /* ========== MUTATIVE FUNCTIONS ========== */
 
-  function stake(uint256 amount)
-    external
+  function stake(uint256 amount) external override {
+    stakeFor(amount, msg.sender);
+  }
+
+  function stakeFor(uint256 amount, address account)
```

Comment:
One security idea: what if on the `stakeFor` function, we added a security requirement that ensures that it is not called by any address except for the vault?

```solidity
require(msg.sender == stakingToken, "Only vault can stake on behalf of users");
```

The require works because it guards the call access to the vault and the msg.sender is coincidentally the `rewardsToken` too. So this prevents mishaps where attackers are able to stake on behalf of other users

Reply:
Yeah, I think this is a good idea. Will add